### PR TITLE
Allow arbitrary brushed motor periods

### PR DESCRIPTION
**Pull request by tridge**
This is a varient of an earlier patch by @khancyr that allows brushed motors to be run with any period

### CODE REVIEW

File: libraries/AP_HAL_PX4/RCOutput.cpp [hunk 1]
```diff
@@ -101,7 +101,9 @@ void PX4RCOutput::_init_alt_channels(void)
 void PX4RCOutput::set_freq_fd(int fd, uint32_t chmask, uint16_t freq_hz, uint32_t &rate_mask) 
 {
     if (_output_mode == MODE_PWM_BRUSHED16KHZ) {
-        freq_hz = 2000; // this maps to 16kHz due to 8MHz clock
+        freq_hz /= 8; // divide by 8 for 8MHz clock
```
Comment (khancyr) [inline libraries/AP_HAL_PX4/RCOutput.cpp]: better 0.125f no?

Comment (EShamaev) [inline libraries/AP_HAL_PX4/RCOutput.cpp]: /= 8 for uint16_t will result in simple shift by compiler

Comment (khancyr) [inline libraries/AP_HAL_PX4/RCOutput.cpp]: oh yes ! forget that freq_hz isn't a float ... sorry !

File: libraries/AP_HAL_PX4/RCOutput.cpp [hunk 2]
```diff
@@ -101,7 +101,9 @@ void PX4RCOutput::_init_alt_channels(void)
 void PX4RCOutput::set_freq_fd(int fd, uint32_t chmask, uint16_t freq_hz, uint32_t &rate_mask) 
 {
     if (_output_mode == MODE_PWM_BRUSHED16KHZ) {
```
Comment (khancyr) [inline libraries/AP_HAL_PX4/RCOutput.cpp]: It shouldn't be _output_mode >= MODE_PWM_BRUSHED16KHZ to allow futur MODE_PWM_BRUSHED ?

File: ArduCopter/system.cpp [hunk 3]
```diff
@@ -685,6 +685,11 @@ void Copter::allocate_motors(void)
         break;
     }
 
+    // brushed 16kHz defaults to 16kHz pulses
+    if (motors->get_pwm_type() >= AP_Motors::PWM_TYPE_BRUSHED16kHz) {
```
Comment (khancyr) [inline ArduCopter/system.cpp]: it should be ==

**Comment by khancyr** [comment]
Agree with @OXINARF

**Pull request merged**